fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/GlobalNetworkManager.java

```diff
@@ -0,0 +1,77 @@
+package com.github.dedis.popstellar.repository.remote;
+
+import androidx.annotation.NonNull;
+
+import com.github.dedis.popstellar.repository.LAORepository;
+import com.github.dedis.popstellar.utility.handler.MessageHandler;
+import com.github.dedis.popstellar.utility.scheduler.SchedulerProvider;
+import com.google.gson.Gson;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.reactivex.disposables.Disposable;
+
+/** Module responsible of managing the connection to the backend server. */
+@Singleton
+public class GlobalNetworkManager implements Disposable {
+
+  private static final String DEFAULT_URL = "ws://10.0.2.2:9000/organizer/client";
+
+  private final LAORepository laoRepository;
+  private final MessageHandler messageHandler;
+  private final ConnectionFactory connectionFactory;
+  private final Gson gson;
+  private final SchedulerProvider schedulerProvider;
+
+  @Nullable private MessageSender networkManager;
+  @Nullable private String currentURL;
+
+  @Inject
+  public GlobalNetworkManager(
+      LAORepository laoRepository,
+      MessageHandler messageHandler,
+      ConnectionFactory connectionFactory,
+      Gson gson,
+      SchedulerProvider schedulerProvider) {
+    this.laoRepository = laoRepository;
+    this.messageHandler = messageHandler;
+    this.connectionFactory = connectionFactory;
+    this.gson = gson;
+    this.schedulerProvider = schedulerProvider;
+
+    connect(DEFAULT_URL);
```

Comment:
does it make sense to have a default connection ? wouldn't it be better to start un-initialized ?

Reply:
Yes, but I would need to work a bit more on the implementation side. So it will be this way for now and I will try to remove this need later.